kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/logs/TestLogCollector.java

```diff
@@ -0,0 +1,232 @@
+ *     │                  ├── pod
+ *     │                  │          ├── describe-pod-kroxylicious-proxy-7db664df48-bxxrx.log
+ *     │                  │          ├── ...
+ *     │                  │          └── logs-pod-kroxylicious-proxy-7db664df48-bxxrx-container-kroxylicious.log
+ *     │                  └── secret
+ *     │                      └── regcred.yaml
+ * ...
+ */
+public class TestLogCollector {
+    private static final String CURRENT_DATE;
+    private final LogCollector logCollector;
+
+    static {
+        // Get current date to create a unique folder
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss");
+        dateTimeFormatter = dateTimeFormatter.withZone(ZoneId.of("GMT"));
+        CURRENT_DATE = dateTimeFormatter.format(LocalDateTime.now());
+    }
+
+    /**
+     * TestLogCollector's constructor
+     */
+    public TestLogCollector() {
+        this.logCollector = defaultLogCollector();
+    }
+
+    /**
+     * Method for creating default configuration of the {@link LogCollector}.
+     * It provides default list of resources into the builder and configures the required {@link KubeClient} and
+     * {@link Kubectl}
+     *
+     * @return  {@link LogCollector} configured with default configuration for the tests
+     */
+    private LogCollector defaultLogCollector() {
+        List<String> resources = new ArrayList<>(List.of(
+                Constants.SECRET.toLowerCase(Locale.ROOT),
+                Constants.DEPLOYMENT.toLowerCase(Locale.ROOT),
+                Constants.CONFIG_MAP.toLowerCase(Locale.ROOT),
+                Constants.SERVICE.toLowerCase(Locale.ROOT),
+                Constants.JOB.toLowerCase(Locale.ROOT),
+                Kafka.RESOURCE_SINGULAR,
+                KafkaNodePool.RESOURCE_SINGULAR));
+
+        return new LogCollectorBuilder()
+                .withKubeClient(new KubeClient())
+                .withKubeCmdClient(new Kubectl())
+                .withRootFolderPath(Environment.CLUSTER_DUMP_DIR)
+                .withNamespacedResources(resources.toArray(new String[0]))
+                .build();
+    }
+
+    /**
+     * Method that checks existence of the folder on specified path.
+     * From there, if there are no sub-dirs created - for each of the test-case run/re-run - the method returns the
+     * full path containing the specified path and index (1).
+     * Otherwise, it lists all the directories, filtering all the folders that are indexes, takes the last one, and returns
+     * the full path containing specified path and index increased by one.
+     *
+     * @param rootPathToLogsForTestCase     complete path for test-class/test-class and test-case logs
+     *
+     * @return  full path to logs directory built from specified root path and index
+     */
+    private Path checkPathAndReturnFullRootPathWithIndexFolder(Path rootPathToLogsForTestCase) {
```

Comment:
the method name isn't too clear.  Maybe something like:

`getSequentiallyNextTargetDirectory.`